enhancements/operator-framework-api/catalogue-check.md

```diff
@@ -0,0 +1,344 @@
+> However, for the checks where the `--optional-values="version=4.x"` which will be using Kubernetes API `1.22` was used we know that the goal is to verify if the operator bundle will work òn `1.22` then, we can already raise the `error` result. 
+
+Then, it means that:
+- `--optional-values="version=value"` flag with a value of an OpenShift version using K8S `=> 1.16 <= 1.22` the validator will return result as warning.
+- `--optional-values="version=value"` flag with a value OpenShift version using K8S `=> 1.22` the validator will return result as error.
+- `minKubeVersion >= 1.22` return the error result.
+- `maxOpenShiftVersion >= OCP version with 1.22` return the error result.
+- `com.redhat.openshift.versions` with a range which has a OCP version with 1.22` return the error result.
+
+**Currently (OCP versions released with the Kubernetes `<1.22`)**
+- `minKubeVersion` and `maxOpenShiftVersion` and `com.redhat.openshift.versions` empty/nil return the warning result.
+
+**After OCP version using the `1.22` Kubernetes release**
+- `minKubeVersion` and `maxOpenShiftVersion` and `com.redhat.openshift.versions` empty/nil return the error result.
+ 
+In a scenario where none version has been provided and the operator bundle has not been using the provided features to define the OpenShift/Kubernetes versions which are supported by it then, it means that the bundle operator should work well in all available possible versions, indeed `>= 1.22` after its release roll out of course. The only check above which will return an error is that one which ensure that the APIs used are supported, however, for this scenario the validator should return **warning** instead.
+
+Note that the OpenShiftValidator needs to be able to receive a string map and check if it contains any valid key for the specific check to address this requirement.
+
+In the code implementation, it should be only one check which will raise error(s) and warning(s) as a respective message according to the OCP/K8S version used. See that the same checks with its specific outputs has been proposed to the `OperatorHubValidator` in the EP [Add new checks to the operatorhub validator][ep-operatorhub-checks]. Therefore, the code could be done without duplicate the rules and instead of that, used for both validators where only type and message changes if possible.
+
+### Risks and Mitigations
+
+- Same risks and/or mitigations described in [Add new checks to the operatorhub validator][ep-operatorhub-checks]
+
+## Design Details
+
+### Open Question
```

Comment:
Hi @kevinrizza @njhale I tried to updated here based on the discussions and I added all these open questions to let us know clarifies our thought an take the decision. I hope that helps. If you have time would be great get your inputs as well.